Add copy mode cursor configuration options

mgpinf opened the pull request:
Add `copy_mode_cursor_fg`, `copy_mode_cursor_bg` configuration options

mgpinf committed:
feat: add copy_mode_cursor configuration options

Add copy_mode_cursor_fg, copy_mode_cursor_bg configuration options

mgpinf committed:
docs: add documentation for copy_mode_cursor configuration options

bew commented on docs/config/appearance.md:
```diff
@@ -71,6 +71,13 @@ config.colors = {
+  -- Overrides the cell background color when copymode is enabled and the current cell is
+  -- occupied by the cursor and the cursor style is set to Block
+  copy_mode_cursor_bg = '#52ad70', -- {{since('nightly', inline=True)}}
+  -- Overrides the text color when copymode is enabled the current cell is occupied by
```
```suggestion
  -- Overrides the text color when copymode is enabled and the current cell is occupied by
```

bew reviewed (changes requested):
Hello!
Adding copy-mode specific handling in termwindow is the wrong way to go, you should instead see how to update the cursor color in the copy-mode overlay

Source code:
https://github.com/wezterm/wezterm/blob/main/wezterm-gui/src/overlay/copy.rs

mgpinf committed:
Update docs/config/appearance.md

Co-authored-by: Benoit de Chezelles <bew@users.noreply.github.com>

mgpinf committed:
fix: update the cursor color in the copy-mode overlay

This change is required because adding copy-mode specific handling in
termwindow is the wrong approach
